cli/sub_command.go

```diff
@@ -456,8 +458,18 @@ func prettyPrintMsg(msg *nats.Msg, headersOnly bool) {
 	}
 
 	if !headersOnly {
-		fmt.Println(string(msg.Data))
-		if !strings.HasSuffix(string(msg.Data), "\n") {
+		data := msg.Data
+		if filter != "" {
+			var err error
+			data, err = filterDataThroughCmd(msg.Data, filter, msg.Subject, "")
```

Comment:
this code is essentially the same just used 3 times, it can probably be made a function.

And improve the error to be more descriptive

Reply:
Well, I try to avoid calling functions when not needed. Of course, one could call it independent of what filter contains and just return data unmodified if no filter is given. I didn't like that, though. Is that what you want?

In my experiments with it, I found the errors that show up satisfying. What is on your mind? Adding something about the error being caused by the translation using a wrapper?

Reply:
I mean this whole section of code the with error handling etc, its all duplicated more or less equally in 3 places.  see after lines 710, 2720 and here, its all very much the same - especially the first two.  I am just curious if it can be extracted since later if someone has to introduce a change in how this work its unlikely they will spot the 3 places thats similar, one might be forgotten and then introduces a bug.

"Error: x" isnt clear, error while doing what?

Reply:
I extended the error to tell where it happens, besides that, there are some minor differences that make it difficult to reduce that code. I would let it as it is right now. Maybe doing some refactoring about outputting the messages in general could be made in another step?

BTW: It just realized that running the template on the individual arguments is not needed, and when done before the `shlex` step, you can even use the template language to run different commands based on the data. I added a new commit that

Reply:
Well. I looked at the code for the output again and see what you mean with the duplication. I guess that can be handled. Adding the "\n" can be done every time. I dislike printing the error from inside the "helper" function. But one could refine that as the `msgOutput()` function and handle the error inside that.